bot/code_review_bot/report/phabricator.py

```diff
@@ -304,95 +319,56 @@ def build_comment(
         former_diff_id=None,
         unresolved=0,
         closed=0,
+        known_issues=[],
     ):
         """
         Build a Markdown comment about published issues
         """
-
-        def pluralize(word, nb):
-            assert isinstance(word, str)
-            assert isinstance(nb, int)
-            return "{} {}".format(nb, nb == 1 and word or word + "s")
-
-        # List all the issues classes
-        issue_classes = {issue.__class__ for issue in issues}
-
-        # Calc stats for issues, grouped by class
-        stats = self.calc_stats(issues)
-
-        # Build parts depending on issues
-        defects, analyzers = set(), set()
-        total_warnings = 0
-        total_errors = 0
-        for stat in stats:
-            defect_nb = []
-            if stat["nb_build_errors"] > 0:
-                defect_nb.append(pluralize("build error", stat["nb_build_errors"]))
-            if stat["nb_defects"] > 0:
-                defect_nb.append(pluralize("defect", stat["nb_defects"]))
-
-            defects.add(
-                " - {nb} found by {analyzer}".format(
-                    analyzer=stat["analyzer"], nb=" and ".join(defect_nb)
-                )
-            )
-            _help = stat.get("help")
-            if _help is not None:
-                analyzers.add(f" - {_help}")
-
-            total_warnings += stat["nb_warnings"]
-            total_errors += stat["nb_errors"]
-
-        # Order both sets
-        defects = sorted(defects)
-        analyzers = sorted(analyzers)
+        comment = ""
 
         # Comment with an absolute link to the revision diff in Phabricator
         # Relative links are not supported in comment and non readable in related emails
         phabricator_diff_url = self.phabricator_diff_url.format(
             diff_id=revision.diff_id
         )
 
-        # Build top comment
-        nb = len(issues)
-
-        # Add extra hint when errors are published outside of the patch
-        defects_details = ""
-        # Only display the hint when the revision has parents in his stack
-        rev_parents_count = len(self.api.load_parents(revision.phabricator_phid))
-        external_failures_count = rev_parents_count and sum(
-            1
-            for issue in issues
-            if issue.is_publishable() and not revision.contains(issue)
-        )
-        if nb == 1 and external_failures_count == 1:
-            defects_details = " (in a parent revision)"
```

Comment:
Is there a way we could detect if the issue was introduced by a parent revision instead of removing support for it?

Reply:
Maybe I'm wrong, but I assumed before/after actually do that, by handling failures outside the diff.
The current code ensures that the Phab revision has parents (I don't know what it exactly means) and count every issue outside the patch as issue from a parent revision.

Reply:
On Phabricator you can have revisions that are linked one another, you can see the whole thing as a PR with multiple commits.
The first commit of the PR is based on a changeset from the repository (in our case, mozilla-central), the other commits of the PR are based on the previous commits of the PR.

Say we have this situation:
- changeset X on mozilla-central
- revision A, based on changeset X
- revision B, based on revision A

When we analyze A, using before/after, we find issues that are new compared to changeset X, and we publish them.
When we analyze B, we will publish the same issues and with this code we were saying that some of them were actually due to A and not B (see https://github.com/mozilla/code-review/pull/1243 and linked issues/PRs). In an ideal world we'd use before/after also between B and A, but that might be too difficult to do now and so we could just keep the message for now.

Reply:
As discussed during today's meeting, handling parent revisions would be more complex (requires to find all the parents revs, then use the endpoint to list existing issues by path and compare them). 
it would be nice to do this in a followup MR though.

Reply:
I opened a new issue, feel free to edit or add details : https://github.com/mozilla/code-review/issues/1705

Reply:
OK, WFM.